src/libs/ajax/leonardo/LeoRuntimesV1DataClient.ts

```diff
@@ -0,0 +1,40 @@
+import { AbortOption, FetchFn } from 'src/libs/ajax/data-client-common';
+import { RawGetRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+
+export interface LeoRuntimesV1DataClient {
+  details: (project: string, name: string, options?: AbortOption) => Promise<RawGetRuntimeItem>;
+  start: (project: string, name: string, options?: AbortOption) => Promise<void>;
+  stop: (project: string, name: string, options?: AbortOption) => Promise<void>;
+}
+
+export interface LeoRuntimesV1DataClientDeps {
+  /**
+   * fetch function that takes care of desired auth/session mechanics, api endpoint root url prefixing,
+   * baseline expected request headers, etc.
+   */
+  fetchAuthedLeo: FetchFn;
+}
+
+const runtimesPath = (project: string, name: string) => `api/google/v1/runtimes/${project}/${name}`;
+
+export const makeLeoRuntimesV1DataClient = (deps: LeoRuntimesV1DataClientDeps): LeoRuntimesV1DataClient => {
```

Comment:
Thoughts on generally making data clients classes?

That could possibly allow a base class to handle adding in the auth header?

Something like:
```ts
class BaseDataClient {
  constructor(authTokenSupplier) { ... }

  authenticatedRequest(url, options) {
    return fetch(url, {
      ...options,
      headers: { ...options.headers, Authorization: `Bearer ${authTokenSupplier()}` }
    });
  }
}

class LeonardoDataClient extends BaseDataClient {
  authenticatedRequest(path, options) {
    return super.authenticatedRequest(`.../${path}`, options);
  }
}
```

Reply:
To summarize what we chatted about on a call, I have come to embrace the elegance of (controlled) functional programming over the many pitfalls of OOP (inheritance and pervasive state-fullness being two big ones).  React hooks gives us a reasonable enough answer on how to handle state in a largely quarantined functional programming wrapper.... so I've largely embraced the FP style in the terra-ui codebase and have largely avoided classes.  Class and interface heavy solutions have their benefits, but also come with a-lot more implementation weight and rigidity.

The chainable fetch function pattern here feels like a nicely incremental solution to 95% remove the concern of auth and root config, etc. from the raw data call authoring that needs to happen in a typical data-client.... if both auth and public fetches are needed, then the depenenency list on the makeCoolDataClient function would expand to include more then one augmented fetch arg.

We can always revisit and iterate later as we learn new things.